packages/components/src/Overlay/OverlaySurface.tsx

```diff
@@ -77,6 +77,8 @@ export const OverlaySurface = forwardRef(
     } = props
     const { closeModal } = useContext(DialogContext)
 
+    useHotkeys('esc', closeModal)
```

Comment:
The nice side effect of these keyboard events being global is that it automatically addresses the issue where you had to focus within a popover before the esc key listener activates

Reply:
I just hope we never have a situation where there is a Popover inside a Dialog or something crazy like that. In that case, pressing `esc` would close everything

Reply:
So we actually have a lot of these because of `Select`. 

I tested against the Select + Dialog example in Storybook (`/?path=/story/forms-select--select-demo`) and the `esc` is definitely a little over-aggressive.

Reply:
Something like this might work (as well as something similar on the Dialog side):
```suggestion    
    const wrapperRef = useRef<HTMLElement>(null)
    const forkedRef = useForkedRef(ref, wrapperRef)

    function handleEsc(e: KeyboardEvent) {
      if (
        wrapperRef.current &&
        wrapperRef.current.contains(document.activeElement)
      ) {
        closeModal()
        e.stopPropagation()
      }
    }
    useHotkeys('esc', handleEsc)
```